mempool/src/core_mempool/transaction.rs

```diff
@@ -59,3 +62,18 @@ pub enum TimelineState {
     // Currently we don't broadcast transactions originated on other peers.
     NonQualified,
 }
+
+#[derive(Clone, Eq, PartialEq)]
+pub struct TransmissionState {
+    pub from_vfn: bool,
+}
+
+impl TransmissionState {
+    pub fn new(from_vfn: bool) -> Self {
+        TransmissionState { from_vfn }
+    }
+
+    pub fn mock() -> Self {
+        Self::new(false)
+    }
+}
```

Comment:
maybe Source? and I guess the point being we don't forward peer to other peers... actually now that I think about it, aren't we also forwarding all validator mempool txns to each other too?